### PR TITLE
wrote tests+resolved queries

### PR DESCRIPTION
**Pull request by sonai99**
Wrote the RTK query for CHALLENGES_URL and replaced the useFetch implementation
Also wrote the test for it

image

image

### CODE REVIEW

**Comment by sonai99** [comment]
had to create a new PR with suggestions from the previous one + wrote the test 
cc @bhtibrewal @Pratiyushkumar @kotesh-arya

File: __tests__/Unit/hooks/challengesApi.test.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        expect(initialResponse.data).toBeUndefined();
+        expect(initialResponse.isLoading).toBe(true);
+
+        await act(() => waitForNextUpdate());
+        const nextResponse = result.current;
+
+        expect(nextResponse.data).not.toBeUndefined();
+        expect(nextResponse.isLoading).toBe(false);
+        expect(nextResponse.isSuccess).toBe(true);
+        console.log(nextResponse.data);
```
Comment (bhtibrewal) [inline __tests__/Unit/hooks/challengesApi.test.tsx:41]: remove this console log

File: src/app/services/challengesApi.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import challengeDataType from '@/interfaces/challenge.type';
+
+type ChallengesQueryResponse = { message: string; challenges: challengeDataType[] };
+
+export const challengesApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    getChallenges: builder.query<ChallengesQueryResponse, void>({
+      query: () => '/challenges',
+      providesTags: ['Challenges'],
+      transformResponse: (response: ChallengesQueryResponse) => {
```
Comment (bhtibrewal) [inline src/app/services/challengesApi.ts:11]: I can't find the sense of transformResponse here if you are sending the response as it is

File: src/pages/challenges/index.tsx [hunk 3]
```diff
@@ -65,7 +59,7 @@ const Challenges: FC = () => {
                     (isLoggedIn ? (
                         isLoading ? (
                             <p>Loading...</p>
-                        ) : error ? (
+                        ) : isError ? (
```
Comment (iamitprakash) [inline src/pages/challenges/index.tsx:62]: can you please change to text like-- "Something Went Wrong, Please Contact Admin !"

File: src/app/services/challengesApi.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ export const challengesApi = api.injectEndpoints({
             query: () => '/challenges',
             providesTags: ['Challenges'],
             transformResponse: (response: ChallengesQueryResponse) => {
-                return response;
+                return response.challenges;
```
Comment (sonai99) [inline src/app/services/challengesApi.ts:15]: @bhtibrewal please review this one

File: src/pages/challenges/index.tsx [hunk 5]
```diff
@@ -54,7 +48,7 @@ const Challenges: FC = () => {
                 setFilteredChallenge(challengeMap);
```
Comment (bhtibrewal) [inline src/pages/challenges/index.tsx:48]: I think it would be if do this filtering in the `tranformResponse` of the RTK Query itself

**Review by bhtibrewal — COMMENTED** [review]
Added a comment everything else seems okay

File: src/app/services/challengesApi.ts [hunk 6]
```diff
@@ -6,13 +6,23 @@ type ChallengesQueryResponse = {
...
             query: () => '/challenges',
             providesTags: ['Challenges'],
             transformResponse: (response: ChallengesQueryResponse) => {
-                return response.challenges;
+                const challenges = response?.challenges;
+                const challengeMap: ChallengeMap = {
+                    Active: challenges?.filter((task) => task.is_active),
+                    Completed: challenges?.filter((task) => !task.is_active),
+                };
+                return challengeMap;
```
Comment (sonai99) [inline src/app/services/challengesApi.ts:25]: @bhtibrewal please check
i have done the filtering in the query itself

**Review by Pratiyushkumar — CHANGES_REQUESTED** [review]
Please write the test cases also

**Comment by sonai99** [comment]
> Please write the test cases also
@Pratiyushkumar 
that i have done in a previous commit only.
Check https://github.com/Real-Dev-Squad/website-status/pull/558/commits/531e9e87ed6a648f9644c9b5012aa2d0e23f043f

**Review by bhtibrewal — DISMISSED** [review]
good work @sonai99  looks okay to me

**Review by Pratiyushkumar — APPROVED** [review]
good work @sonai99, looks good to me

**Comment by sonai99** [comment]
@bhtibrewal I have resolved the build fail
It was due to a check missing in the rtk query.

**Review by bhtibrewal — APPROVED** [review]
Looks okay to me

**Comment by sonai99** [comment]
@bhtibrewal can you please have a look at the latest commit.
Last time build failed due to the test failing, i have corrected it and its passing on local now.

**Comment by sonai99** [comment]
@Pratiyushkumar Could you please have a look at this, its still awaiting approval. Not sure if i am required to do anything more on this.

**Review by Pratiyushkumar — DISMISSED** [review]

**Comment by sonai99** [comment]
@Pratiyushkumar i have resolved the queries

**Review by Pratiyushkumar — DISMISSED** [review]
LGTM

**Comment by sonai99** [comment]
@kotesh-arya  / @shubhamsigdar1 can you please review this, need one more approval, its getting really delayed.

**Review by Pratiyushkumar — APPROVED** [review]
LGTM

**Comment by sonai99** [comment]
@iamitprakash  / @bhtibrewal  
General reminder that this PR is still awaiting approval.

**Comment by sonai99** [comment]
@bhtibrewal  any reasons this hasn't been approved yet?
I got the approval from @Pratiyushkumar, just need one more.

**Review by bhtibrewal — APPROVED** [review]
LGTM, nice work @sonai99

**Pull request merged**